src/ucs/async/signal.c

```diff
@@ -445,7 +445,7 @@ ucs_async_signal_timerq_add_timer(ucs_async_signal_timer_t *timer, int tid,
 
     }
 
-    status = ucs_timerq_add(&timer->timerq, timer_id, interval);
+    status = ucs_timerq_add(&timer->timerq, *timer_id_p, interval);
```

Comment:
so this function now uses timer_id_p as INOUT instead of OUT?
maybe add the minimal code in timerq to generate an id (e.g `*timer_id_p = timerq->num_timers++`) so no need to set  handler->timer_id = handler->id before calling this function?

Reply:
@yosefe `ucs_async_signal_timerq_add_timer` is now using `timer_id_p` as INOUT, but `ucs_timerq_add` is still using `timer_id` as IN... so we can either change the API of `ucs_timerq_add` to add this code there, or leave it in `ucs_async_alloc_handler`.

Reply:
can we change the api of ucs_timerq_add to return *timer_id_p as OUT (trivial id allocation e.g `*timer_id_p = timerq->num_timers++`) ?

Reply:
Sure, will do.